activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -141,6 +141,10 @@ def supports_datetime_with_precision?
         end
       end
 
+      def supports_virtual_columns?
+        !mariadb? && version >= "5.7.5"
```

Comment:
I use MySQL locally and MariaDB in production, so this setup would work for me.

I think it is pretty confusing for people doing development against MariaDB locally, though. Running tests dumps dev schema to load for the test db, so your test runs would fail with a schema you can't round-trip :(

People could still use SQL dumps for schema and use this declarative style for migrations, however.

What route do you think is friendliest and most useful to people doing local development with MariaDB?